airflow/jobs/base_job.py

```diff
@@ -21,6 +21,8 @@
 from time import sleep
 from typing import Optional
 
+from pendulum import utcfromtimestamp
+from redis import Redis
```

Comment:
Will this mean a hard requirement on Redis for all ?

Can we make it optional import